packages/web/hooks/window/use-query-param-state.ts

```diff
@@ -11,8 +11,10 @@ type QueryParamValue =
   | null
   | undefined;
 
-/** Emulates `React.useState` but uses next/router to store state in URL query params.
- *  Only sets from `defaultValue` if it's not present in the query params. */
+/**
+ * Emulates `React.useState` but uses next/router to store state in URL query params.
+ * Only sets from `defaultValue` if it's not present in the query params.
+ */
 export function useQueryParamState<TValue extends QueryParamValue>(
```

Comment:
Instead of using `useEffect` and manually manage query params, inside the pools table we're already using [`nuqs`](https://nuqs.47ng.com/docs/installation), do you think we can use it here too? It can simplify the code a bit, but mostly it avoids unnecessary rerendering

Reply:
Thank you for the suggestion! To avoid expanding the scope and delaying the current PR, I have created an [issue](https://linear.app/osmosis/issue/FE-197/sync-swap-tools-token-query-params-with-nuqs) in Linear to address this in a separate PR.